src/richie/apps/core/serializers.py

```diff
@@ -8,12 +9,29 @@ class UserSerializer(ModelSerializer):
     """Serializer for User objects."""
 
     full_name = SerializerMethodField()
+    urls = SerializerMethodField()
 
     class Meta:
         model = get_user_model()
-        fields = ["full_name", "username"]
+        fields = ["full_name", "urls", "username"]
 
     @staticmethod
     def get_full_name(user):
         """Get the user's full_name on the User class during serialization."""
         return user.get_full_name()
+
+    @staticmethod
+    def get_urls(user):
+        """Get the user's urls related to LMS"""
+
+        def format_link(link):
+            """Bind base_url and username into link"""
+            return {
+                key: value.format(
+                    base_url=settings.LMS_BACKENDS[0]["BASE_URL"],
+                    username=user.username,
+                )
+                for key, value in link.items()
+            }
+
+        return map(format_link, settings.MAIN_LMS_USER_URLS)
```

Comment:
It should be the same as:
```suggestion
            return {
                key: value.format(
                    base_url=settings.LMS_BACKENDS[0]["BASE_URL"],
                    username=user.username,
                )
                for key, value in settings.MAIN_LMS_USER_URLS.items()
            }
```

Reply:
I used `map` because `MAIN_LMS_USER_URLS` is a List of  Dicts. I would like translate link labels from the backend. In this way, frontend is totally agnostic about what it has to render.

```python
MAIN_LMS_USER_URLS = values.ListValue(
        [
            {"label": _("Profile"), "href": _("{base_url:s}/u/{username:s}")},
            {"label": _("Account"), "href": _("{base_url:s}/account/settings")},
        ]
    )
```

Reply:
I think @sampaccoud mostly means using a list comprehension instead of a `map` call so it's more idiomatic python

```python
return [
  {
    key: value.format(
      base_url=settings.LMS_BACKENDS[0]["BASE_URL"],
      username=user.username,
    )
    for key, value in link.items()
  }
  for link in settings.MAIN_LMS_USER_URLS
]
```